lib/dns.js

```diff
@@ -82,11 +82,36 @@ function onlookup(err, addresses) {
 }
 
 
+function onlookupall(err, addresses) {
+  var i, results = [];
+  if (err) {
+    return this.callback(errnoException(err, 'getaddrinfo', this.hostname));
+  }
+
+  for (i = 0; i < addresses.length; i++) {
+    if (this.family) {
```

Comment:
I think this can be removed if you do `family: this.family || (addresses[i].indexOf(':') >= 0 ? 6 : 4)`